### DISCUSSION TITLE
Unknown server message with code 130

### DISCUSSION THREAD

**Discussion by mathiascode** [jpdillingham/Soulseek.NET#574] [General]
A Nicotine+ user received a server message with code 130 earlier today. The message is empty, and was sent six times:

```
[2021-08-24 17:37:15] Server message type 130 size 0 contents bytearray(b'') unknown
[2021-08-24 17:37:15] Server message type 130 size 0 contents bytearray(b'') unknown
[2021-08-24 17:37:15] Server message type 130 size 0 contents bytearray(b'') unknown
[2021-08-24 17:37:15] Server message type 130 size 0 contents bytearray(b'') unknown
[2021-08-24 17:37:15] Server message type 130 size 0 contents bytearray(b'') unknown
[2021-08-24 17:37:15] Server message type 130 size 0 contents bytearray(b'') unknown
```

Looking at the SoulseekQt message name list, this seems to be a "DNet reset" message. My theory is that this message could be used to reset peers with strange data in the distributed network, since the Nicotine+ user sent an invalid branch level to the server due to a recent Nicotine+ regression, but I haven't been able to reproduce this. I don't recall seeing this message previously either.

Based on the message name and lack of contents in the message, I've made it unset the active parent and request a new one in Nicotine+: https://github.com/nicotine-plus/nicotine-plus/commit/a40de24e50d41612081900f522d8dda0dbe36413

If you manage to figure out more about this message, let me know. :)

**Comment by jpdillingham** [comment]
Thanks for the heads up.  I'm wondering if this also might be sent if your speed and the speed of your parent diverge by too much, since the server seems to care a lot about that.

Either way I agree that this should cause the current parent to be dropped.  I wonder if we should disconnect children also?

**Reply by jpdillingham** [reply]
Although, I guess the server would send the children the same message if they needed to be reset.  So probably not necessary.

**Comment by jpdillingham** [comment]
There was further speculation that this might be sent if the depth of the distributed network gets too deep in #641
